File: gaia-sdk-javascript/src/graphql/request/type/SkillIntrospection.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";

export class SkillIntrospection extends Array<(_:VariableRegistry) => string> {

    public name = () => { 
        this.push(_ => "name")
    };

    public state = () => { 
        this.push(_ => "state")
    };

    public started = () => { 
        this.push(_ => "started")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/SkillIntrospection.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class SkillIntrospection(list):

    def name(self):
        self.append(lambda x: "name")

    def state(self):
        self.append(lambda x: "state")

    def started(self):
        self.append(lambda x: "started")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
